src/kernel/arch/x86_64/cpu/tss.c

```diff
@@ -0,0 +1,65 @@
+#include <logging.h>
+#include <tss.h>
+
+extern uint64_t gdt64[];
+extern uint64_t stack;
+
+tss_t kernel_tss;
+
+void initialize_tss(){
+
+    loglinef(Verbose, "(%s) Initializing tss", __FUNCTION__);
+
+    // These fields are reserved and must be set to 0
+    kernel_tss.reserved0 = 0x00;
```

Comment:
I like that this is all explicit, but it'd be more efficient to use `memset()` to zero the whole tss and then set the fields you want afterwards.

Reply:
yeah, i totally agree, but i haven't implemented memset yet XD